src/Explorer/Panes/UploadItemsPane/index.tsx

```diff
@@ -0,0 +1,220 @@
+import { GenericRightPaneComponent, GenericRightPaneProps } from "../GenericRightPaneComponent";
+import { Tooltip } from "./Tooltip";
+
+const UPLOAD_FILE_SIZE_LIMIT = 2097152;
+
+export interface UploadItemsPaneProps {
+  explorer: Explorer;
+  closePanel: () => void;
+}
+
+interface IUploadFileData {
+  numSucceeded: number;
+  numFailed: number;
+  fileName: string;
+}
+
+const getTitle = (): string => {
+  if (userContext.apiType === "Cassandra" || userContext.apiType === "Tables") {
+    return "Upload Tables";
+  } else if (userContext.apiType === "Gremlin") {
+    return "Upload Graph";
+  } else {
+    return "Upload Items";
+  }
+};
+
+export const UploadItemsPane: FunctionComponent<UploadItemsPaneProps> = ({
+  explorer,
+  closePanel,
+}: UploadItemsPaneProps) => {
+  const [selectedFilesTitle, setSelectedFilesTitle] = useState<string>("");
+  const [files, setFiles] = useState<FileList>();
+  const [uploadFileDataVisible, setUploadFileDataVisible] = useState<boolean>(false);
+  const [uploadFileData, setUploadFileData] = useState<UploadDetailsRecord[]>([]);
+  const [formError, setFormError] = useState<string>("");
+  const [formErrorDetail, setFormErrorDetail] = useState<string>("");
+  const [isExecuting, setIsExecuting] = useState<boolean>();
+
+  const fileRef = useRef<HTMLInputElement>();
+  useEffect(() => {
+    setUploadFileDataVisible(uploadFileData?.length > 0);
+  }, [uploadFileData]);
+
+  const onSubmit = () => {
+    setFormError("");
+    if (!files || files.length === 0) {
+      setFormError("No files specified");
+      setFormErrorDetail("No files were specified. Please input at least one file.");
+      logConsoleError("Could not upload items -- No files were specified. Please input at least one file.");
+    } else if (_totalFileSizeForFileList(files) > UPLOAD_FILE_SIZE_LIMIT) {
+      setFormError("Upload file size limit exceeded");
+      setFormErrorDetail("Total file upload size exceeds the 2 MB file size limit.");
+      logConsoleError("Could not upload items -- Total file upload size exceeds the 2 MB file size limit.");
+    }
+
+    const selectedCollection = explorer.findSelectedCollection();
+
+    setIsExecuting(true);
+
+    selectedCollection
+      ?.uploadFiles(files)
+      .then(
+        (uploadDetails: UploadDetails) => {
+          setUploadFileData(uploadDetails.data);
+          setSelectedFilesTitle("");
+          setFiles(undefined);
+        },
+        (error: Error) => {
+          const errorMessage = getErrorMessage(error);
+          setFormError(errorMessage);
+          setFormErrorDetail(errorMessage);
+        }
+      )
+      .finally(() => {
+        setIsExecuting(false);
+      });
+  };
+
+  const updateSelectedFiles = (event: ChangeEvent<HTMLInputElement>): void => {
+    const { files: newFiles } = event.target;
+
+    let newFileList = "";
+
+    if (!newFiles || newFiles.length === 0) {
+      return;
+    }
+
+    for (let i = 0; i < newFiles?.length; i++) {
```

Comment:
If we check that `newFiles` exists in line 93, I suggest we drop the optional `?`: `newFiles.length`.